addons/test_mail/models/test_mail_models.py

```diff
@@ -37,6 +37,7 @@ class MailTestActivity(models.Model):
     _inherit = ['mail.thread', 'mail.activity.mixin']
 
     name = fields.Char()
+    date = fields.Date()
```

Comment:
What is the impact of this new date field ? Looking at code, I am not sure to understand its purpose.

Reply:
The optimization of `_read_progress_bar` for `activity_state` must deal with aggregations and return results in the right format.
Grouping by `date:week` checks that the code aggregates the field as expected (`date_trunc(...)`), formats it as expected (`W42 2021`) and returns it with the expected key (`date:week`).